app/src/main/java/com/parishod/watomatic/NotificationService.java

```diff
@@ -40,6 +39,32 @@ private boolean canReply(StatusBarNotification sbn){
                 canSendReplyNow(sbn);
     }
 
+    private boolean shouldReply(StatusBarNotification sbn){
+        PreferencesManager prefs = PreferencesManager.getPreferencesInstance(this);
+        boolean isGroup = sbn.getNotification().extras.getBoolean("android.isGroupConversation");
```

Comment:
Everything is good except group detection. This should work ideally but there is an issue where images messages in group are not being detected as `isGroupConversation` sometimes (#181).

So there is an extra logic to detect that: [src](https://github.com/adeekshith/watomatic/blob/eac18a2877718eb3a2f07493f91b8d6dab250900/app/src/main/java/com/parishod/watomatic/NotificationService.java#L149-L156)

May be we should move that logic to a separate method and reuse it.
Or I will merge this PR as is this evening and raise another one fixing this.

Reply:
I tried using that logic, but didn't manage to do it right, so went the easy way as a temporary fix.

Reply:
I've read the issue you mentioned, if we are gonna add a new option to do extra checks in settings, i think this should be changed on the new PR

Reply:
No probs 😊
I'll merge this PR in the evening ([CST](https://www.wolframalpha.com/input/?i=Current+time+in+lllinois%2C+USA)) as is and make that fix before I forget. Great work!! 🙌

Also, as you are already good with Kotlin, feel free to use that in the future (anything is fine though). I am planning to move this project to Kotlin in the near future anyways as it is getting difficult to stick to Java in the Android world.

Reply:
> I've read the issue you mentioned, if we are gonna add a new option to do extra checks in settings, i think this should be changed on the new PR

Yep, Had to do a hotfix for a previous release. As soon as that is done, will make a PR to fix this group issue.

Reply:
> No probs blush
I'll merge this PR in the evening (CST) as is and make that fix before I forget. Great work!! raised_hands

Sounds nice

> Also, as you are already good with Kotlin, feel free to use that in the future (anything is fine though). I am planning to move this project to Kotlin in the near future anyways as it is getting difficult to stick to Java in the Android world.

Yay!